lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp

```diff
@@ -1242,6 +1334,109 @@ bool SwiftASTManipulator::AddExternalVariables(
+                                                                     swift::SourceLoc(),
+                                                                     0,
+                                                                     generic_param_index);
+      generic_param_index++;
+      generic_param_decls.push_back(parameter);
+      auto interface_type = parameter->getInterfaceType()->getMetatypeInstanceType().getPointer();
+      if (auto new_generic = llvm::dyn_cast<swift::GenericTypeParamType>(interface_type)) {
+        generic_param_types.push_back(new_generic);
+      }
+    }
+  }
+  auto inner_parameters = swift::ParameterList::create(ast_context,
+                                                       l_paren_loc,
+                                                       param_decls,
+                                                       r_paren_loc);
+  m_inner_function_decl->setName(swift::DeclName(ast_context,
+                                                 m_inner_function_decl->getBaseName(),
+                                                 inner_parameters));
+  m_inner_function_decl->setParameters(inner_parameters);
+
+  auto signature = swift::GenericSignature::get(
+      generic_param_types, swift::ArrayRef<swift::Requirement>());
+  m_inner_function_decl->setGenericSignature(signature);
+
+  auto generic_param_list = swift::GenericParamList::create(ast_context,
+                                                            l_paren_loc,
+                                                            generic_param_decls,
+                                                            r_paren_loc);
+  m_inner_function_decl->setGenericParams(generic_param_list);
+
+  class UpdateCallToInnerWalker : public swift::ASTWalker {
+  private:
+    swift::ASTContext &m_ast_context;
+    llvm::ArrayRef<swift::Expr *> m_decl_refs;
+    llvm::ArrayRef<swift::Identifier> m_names;
+  public:
+    swift::CallExpr *call = nullptr;
+    UpdateCallToInnerWalker(swift::ASTContext &ast_context,
+             llvm::ArrayRef<swift::Expr *> decl_refs,
+             llvm::ArrayRef<swift::Identifier> names)
+        : m_ast_context(ast_context),
+          m_decl_refs(decl_refs),
+          m_names(names) {}
+
+    std::pair<bool, swift::Expr *> walkToExprPre(swift::Expr *E) override {
+      if (auto call_expr = llvm::dyn_cast<swift::CallExpr>(E)) {
+        auto new_call = swift::CallExpr::createImplicit(m_ast_context,
+                                                        call_expr->getFn(),
+                                                        m_decl_refs,
+                                                        m_names);
+        call = new_call;
+        return {false, new_call};
+      }
+      return ASTWalker::walkToExprPre(E);
+    }
+  };
+  UpdateCallToInnerWalker walker(ast_context,
+                                 decl_refs,
+                                 names);
+  m_function_decl->walk(walker);
```

Comment:
It's usually helpful to call `swift::verify(Decl)` after modifying a decl (maybe just in Asserts mode), this can flag issues with the AST that are otherwise hard-to-diagnose.